docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,956 @@
+    }
+
+    // override the _setData function so that the LSP8ReferenceContract is not editable
+    function _setData(
+        bytes32 dataKey,
+        bytes memory dataValue
+    ) internal override {
+        require(
+            dataKey != _LSP8_REFERENCE_CONTRACT,
+            "LSP8ReferenceContractNotEditable"
+        );
+        LSP4DigitalAssetMetadata._setData(dataKey, dataValue);
+    }
+}
+
+```
+
+</details>
+
+## Create the LSP8 token
+
+Now that we have the LSP7 token, we can create the LSP8 token.
+
+### Specify what the LSP8 token will need to do
+
+The LSP8 token will be in charge of deploying the LSP7 token and minting the initial supply. When minting the LSP7 tokens, the deployed LSP7 token will be set as a `tokenId` of the LSP8 token.
+For this reason, when minting a new token on the LSP8 contract (a new LSP7 token contract), we will need to set the following:
+
+- the `name` of the LSP7 token
+- the `symbol` of the LSP7 token
+- the [`lsp4TokenType`](../../standards/tokens/LSP7-Digital-Asset.md##lsp4tokentype) of the LSP7 token
+- the `isNonDivisible` of the LSP7 token
+- the `totalSupply` of the LSP7 token
+- the `receiver` of the initial tokens of the LSP7 token
+
+### Imports
+
+Let's start creating an `LSP8Token.sol` file in the `contracts` folder and import:
+
+- the [`LSP8IdentifiableDigitalAsset.sol`](https://github.com/lukso-network/lsp-smart-contracts.git) from which we will inherit the main functionalities of the LSP8
+- the [`_LSP4_METADATA_KEY`](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata) constant
+- the LSP7 token contract we just created so that we can use it in our `mint(...)` function
+
+```typescript
+import { LSP8IdentifiableDigitalAsset } from '@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.sol';
+import { _LSP4_METADATA_KEY } from '@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4Constants.sol';
+import { LSP7Token } from './LSP7Token.sol';
+```
+
+### Contract
+
+#### Constructor
+
+The constructor will be in charge of setting the following parameters:
+
+- `name`: the name of the LSP8 token
+- `symbol`: the symbol of the LSP8 token
+- `owner`: the owner of the LSP8 token that will be able change the contract metadata and mint new LSP7 tokens
+- [`lsp4TokenType`](../../standards/tokens/LSP7-Digital-Asset.md##lsp4tokentype) of the LSP7 token
+- `lsp8TokenIdFormat`: the [format of the tokenId](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenidformat) of the LSP8 token
+- `lsp4MetadataURI`: the [LSP4 Metadata](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata)
+
+```typescript
+contract LSP8Token is LSP8IdentifiableDigitalAsset {
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address newOwner_,
+        uint256 lsp4TokenType_,
+        uint256 lsp8TokenIdFormat_,
```

Comment:
Since this tutorial uses `Address` as the `lsp8TokenIdFormat`, can't we pass that as a parameter imported from the `LSP8Constants.sol`?

```solidity
import { _LSP8_TOKENID_FORMAT_ADDRESS } from "@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8Constants.sol";

constructor(
        string memory name_,
        string memory symbol_,
        address newOwner_,
        uint256 lsp4TokenType_,
        bytes memory lsp4MetadataURI_
    )
        LSP8IdentifiableDigitalAsset(
            name_,
            symbol_,
            newOwner_,
            lsp4TokenType_,
            _LSP8_TOKENID_FORMAT_ADDRESS
        )
    {
        // set the lsp4MetadataURI
        _setData(_LSP4_METADATA_KEY, lsp4MetadataURI_);
    }
// ...
```

Reply:
good point